File: Tutoriels/CSharp/Compilation/Reference_Creation/src/GMath.cs
//===============================================
using System;
//===============================================
public class GMath {
    //===============================================
    private GMath() {
        
    }
    //===============================================
    public static GMath Instance() {
        lock(m_instanceLock) {
            if(m_instance == null) {
                m_instance = new GMath();
            }
            return m_instance;
        }
    }
    //===============================================
    public double Add(double a, double b) {
        double m_data = a + b;
        return m_data;
    }
    //===============================================
    public double Sub(double a, double b) {
        double m_data = a - b;
        return m_data;
    }
    //===============================================
    public double Mul(double a, double b) {
        double m_data = a * b;
        return m_data;
    }
    //===============================================
    public double Div(double a, double b) {
        double m_data = a / b;
        return m_data;
    }
    //===============================================
    private static GMath m_instance = null;
    private static readonly object m_instanceLock = new object();
    //===============================================
}
//===============================================

File: Tutoriels/Cpp/Make/code/Librairie_Statique_Creation/src/manager/GMath.cpp
//===============================================
#include "GMath.h"
//===============================================
GMath* GMath::m_instance = 0;
//===============================================
GMath::GMath() {

}
//===============================================
GMath::~GMath() {

}
//===============================================
GMath* GMath::Instance() {
    if(m_instance == 0) {
        m_instance = new GMath;
    }
    
    return m_instance;
}
//===============================================
double GMath::add(const double& a, const double& b) {
    double m_data = a + b;
    return m_data;
}
//===============================================
double GMath::subtract(const double& a, const double& b) {
    double m_data = a - b;
    return m_data;
}
//===============================================
double GMath::multiply(const double& a, const double& b) {
    double m_data = a * b;
    return m_data;
}
//===============================================
double GMath::divide(const double& a, const double& b) {
    double m_data = a / b;
    return m_data;
}
//===============================================
double GMath::quotient(const double& a, const double& b) {
    double m_data = floor(divide(a, b));
    return m_data;
}
//===============================================
double GMath::modulo(const int& a, const int& b) {
    int m_data = a % b;
    return double(m_data);
}
//===============================================
